hw/xbox/smbus_xbox_smc.c

```diff
@@ -178,9 +180,41 @@ static uint8_t smc_receive_byte(SMBusDevice *dev)
     return 0;
 }
 
+bool smbus_avpack_to_reg(char *avpack, uint8_t *value)
+{
+    uint8_t r;
+
+    if (!strlen(avpack)) {
+        /* pretend to have a composite av pack plugged in by default */
+        r = SMC_REG_AVPACK_COMPOSITE;
```

Comment:
This code path is not good, because it prevents re-using this function for the monitor. We probably don't want a monitor command with empty argument to switch to composite.

@mborgerson and myself discussed this on XQEMU IRC; I'll mark what I believe was the solution we had agree'd on instead.

Reply:
Why in the world we need to reuse avpack function for the monitor, is there any specific reason for that?

Reply:
Because you can physically unplug / replug an AV cable at runtime on a real console. The [SMC will actually handle this case explicitly and signal an interrupt](http://xboxdevwiki.net/PIC#Interrupt_reasons).

I'm not sure whose responsibility it is (kernel or XDK), but most programs will probably only reboot the Xbox. However, we should still support it, as it's an option that people have with a physical Xbox, too.

The monitor command will be added in a future PR, but we should still prepare for it. Hence this helper function in the first place.

Reply:
Improved.

Reply:
Just a minor nit: `if (!strlen(avpack))` can be optimized by checking the first character of `avpack` if it is `0` via `if (!avpack[0])` or `if (!*avpack)` (what I do) & save yourself a call to `strlen()`.

This can break if the `avpack` pointer is `NULL` & I also sanity check for that.

Reply:
It also makes it harder to read in a non-hot code path, for a case that would likely be optimized anyway (regardless of how the check is written).